OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs

```diff
@@ -15,124 +18,101 @@ namespace OutOfSchool.WebApi.Services
+        private readonly ILogger logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ChildService"/> class.
         /// </summary>
         /// <param name="repository">Repository for the Child entity.</param>
-        public ChildService(IEntityRepository<Child> repository)
+        /// <param name="logger">Logger.</param>
+        public ChildService(IEntityRepository<Child> repository, ILogger logger)
         {
             this.repository = repository;
+            this.logger = logger;
         }
 
         /// <inheritdoc/>
         public async Task<ChildDTO> Create(ChildDTO dto)
         {
-            if (dto == null)
-            {
-                throw new ArgumentNullException(nameof(dto), "Child was null.");
-            }
-
-            if (dto.DateOfBirth > DateTime.Now)
-            {
-                throw new ArgumentException("Invalid Date of birth");
-            }
+            logger.Information("Child creating was started.");
 
-            try
-            {
-                var child = dto.ToDomain();
+            var child = dto.ToDomain();
 
-                var newChild = await repository.Create(child).ConfigureAwait(false);
+            var newChild = await repository.Create(child).ConfigureAwait(false);
 
-                return newChild.ToModel();
-            }
-            catch (Exception ex)
-            {
-                throw new Exception($"{nameof(dto)} could not be saved: {ex.Message}");
-            }
+            return newChild.ToModel();
         }
 
         /// <inheritdoc/>
         public async Task<IEnumerable<ChildDTO>> GetAll()
         {
-            try
-            {
-                var children = await repository.GetAll().ConfigureAwait(false);
-               
-                return children.Select(child => child.ToModel()).ToList();
-            }
-            catch (Exception ex)
-            {
-                throw new Exception($"Couldn't retrieve Children: {ex.Message}");
-            }
+            logger.Information("Process of getting all Children started.");
+
+            var children = await repository.GetAll().ConfigureAwait(false);
+
+            logger.Information(!children.Any()
+                ? "Child table is empty."
+                : "Successfully got all records from the Child table.");
+
+            return children.Select(x => x.ToModel()).ToList();
         }
 
         /// <inheritdoc/>
         public async Task<ChildDTO> GetById(long id)
         {
+            logger.Information("Process of getting Child by id started.");
+
             var child = await repository.GetById(id).ConfigureAwait(false);
 
             if (child == null)
             {
-                throw new ArgumentException("Incorrect Id!", nameof(id));
+                throw new ArgumentOutOfRangeException(id.ToString(),
+                    "The id cannot be greater than number of table entities.");
```

Comment:
maybe child is null because of some other reason, the message "The id cannot be greater than number of table entities." might be confusing

Reply:
I cannot find another reason that could cause an exception here. 
The first check is located in the controller - it checks param if it is not less than 1. 
Then another one is in the service - if there is no entity found by this id, for me it just means that id is greater than entities count, and we get null. 
Can’t be null in the database?

Reply:
if we retrieve a child that was previously deleted, we'll get null but error message won't be relevant